network/src/legacy/gossip/mod.rs

```diff
@@ -334,11 +334,19 @@ pub fn register_validator<C: ChainContext + 'static, S: NetworkSpecialization<Bl
 	let gossip_side = validator.clone();
 	let gossip_engine = sc_network_gossip::GossipEngine::new(
 		service.clone(),
-		executor,
 		POLKADOT_ENGINE_ID,
 		gossip_side,
 	);
 
+	// Ideally this would not be spawned as an orphaned task, but polled by
+	// `RegisteredMessageValidator` which in turn would be polled by a `ValidationNetwork`.
+	let spawn_res = executor.spawn_obj(futures::task::FutureObj::from(Box::new(gossip_engine.clone())));
+
+	// Note: we consider the chances of an error to spawn a background task almost null.
```

Comment:
I agree, but should we panic instead? This code will only be run on node startup, maybe it makes sense to just fail right away?

Reply:
A graceful exit would be hard to include, but maybe a warning would be good.